### PR TITLE
ChargeItem search by context changes

### PR DESCRIPTION
**Pull request by Anurag-Balannagari**
Description
----
Adding documentation for ChargeItem Search by Context.
Screen Shot 2021-06-14 at 10 47 15 AM
Screen Shot 2021-06-14 at 4 37 26 PM
Screen Shot 2021-06-14 at 10 48 13 AM

PR Checklist
----
- [x] Screenshot(s) of changes attached before changes merged.
- [ ] Screenshot(s) of changes attached after changes merged and published.

### CODE REVIEW

File: content/millennium/r4/financial/charge-item.md [hunk 1]
```diff
@@ -86,10 +86,15 @@ Search for ChargeItems that meet supplied query parameters:
...
  `_id`                 | Yes        | [`token`]     | The logical resource id associated with the ChargeItem.
+ `context`             | Yes        | [`token`]     | Encounter associated with event.
+[`_count`]             | No         | [`number`]    | The maximum number of results to return.
+  `-status`            | No         | [`token`]     | The status of the encounter. Example: `billable`
 
 Notes:
 
 - When searching by `_id`, only a single ChargeItem id can be provided.
+- `_count` and `-status` must be provided with context.
+- `-status` valid parameter is `billable`
```
Comment (Coffeevee) [inline content/millennium/r4/financial/charge-item.md:97]: let's reword this to say something like:
`-status` must be set to `billable`

File: content/millennium/r4/financial/charge-item.md [hunk 2]
```diff
@@ -86,10 +86,15 @@ Search for ChargeItems that meet supplied query parameters:
  Name                  | Required?  | Type          | Description
 -----------------------|------------|---------------|-------------------------------------------------------
  `_id`                 | Yes        | [`token`]     | The logical resource id associated with the ChargeItem.
```
Comment (alexbeckner1) [inline content/millennium/r4/financial/charge-item.md:88]: Instead of Requrred? 'Yes' for _id and context, should we have 'This or context' for id and 'This or _id' for context?

File: content/millennium/r4/financial/charge-item.md (hunk 1 above)
Comment (alexbeckner1) [reply content/millennium/r4/financial/charge-item.md:97]: +1 to rewording

File: content/millennium/r4/financial/charge-item.md (hunk 2 above)
Comment (Anurag-Balannagari) [reply content/millennium/r4/financial/charge-item.md]: updated here https://github.com/cerner/fhir.cerner.com/pull/665/commits/aa37a5294afd32516a75bd83c1abd309654fb0e9#diff-16d4f8f74626c62d57a8b71d6be699f60e72fffed8636bc32e6654d4252b3ebbR88-R89

File: content/millennium/r4/financial/charge-item.md (hunk 1 above)
Comment (Anurag-Balannagari) [reply content/millennium/r4/financial/charge-item.md]: updated here https://github.com/cerner/fhir.cerner.com/pull/665/commits/aa37a5294afd32516a75bd83c1abd309654fb0e9#diff-16d4f8f74626c62d57a8b71d6be699f60e72fffed8636bc32e6654d4252b3ebbR97

File: content/millennium/r4/financial/charge-item.md (hunk 2 above)
Comment (alexbeckner1) [reply content/millennium/r4/financial/charge-item.md]: +1

File: content/millennium/r4/financial/charge-item.md (hunk 1 above)
Comment (alexbeckner1) [reply content/millennium/r4/financial/charge-item.md]: nit: can we add a period to the end of it to match the other notes

Comment (Anurag-Balannagari) [reply content/millennium/r4/financial/charge-item.md]: updated here https://github.com/cerner/fhir.cerner.com/pull/665/commits/6782742b24d6e089283790fdc2d3280dc576d991#diff-16d4f8f74626c62d57a8b71d6be699f60e72fffed8636bc32e6654d4252b3ebbR97

**Comment by Coffeevee** [comment]
Friendly reminder to update screenshots before sending this out for FHIR review :)

**Review by Coffeevee — APPROVED** [review]

File: content/millennium/r4/financial/charge-item.md [hunk 3]
```diff
@@ -83,13 +83,18 @@ Search for ChargeItems that meet supplied query parameters:
...
 
- Name                  | Required?  | Type          | Description
------------------------|------------|---------------|-------------------------------------------------------
- `_id`                 | Yes        | [`token`]     | The logical resource id associated with the ChargeItem.
+ Name                  | Required?       | Type          | Description
+-----------------------|-----------------|---------------|-------------------------------------------------------
+ `_id`                 | This or context | [`token`]     | The logical resource id associated with the ChargeItem.
+ `context`             | This or _id     | [`token`]     | Encounter associated with event.
+[`_count`]             | No              | [`number`]    | The maximum number of results to return.
+  `-status`            | No              | [`token`]     | The status of the encounter. Example: `billable`
```
Comment (exopsRamba) [inline content/millennium/r4/financial/charge-item.md:91]: Should we state `See notes` for the -status Required? column? We do something similar for Account type since it's conditionally required. 

https://github.com/cerner/fhir.cerner.com/blame/main/content/millennium/r4/financial/account.md#L80

File: content/millennium/r4/financial/charge-item.md [hunk 4]
```diff
@@ -83,13 +83,18 @@ Search for ChargeItems that meet supplied query parameters:
...
 ### Parameters
 
- Name                  | Required?  | Type          | Description
------------------------|------------|---------------|-------------------------------------------------------
- `_id`                 | Yes        | [`token`]     | The logical resource id associated with the ChargeItem.
+ Name                  | Required?       | Type          | Description
+-----------------------|-----------------|---------------|-------------------------------------------------------
+ `_id`                 | This or context | [`token`]     | The logical resource id associated with the ChargeItem.
+ `context`             | This or _id     | [`token`]     | Encounter associated with event.
+[`_count`]             | No              | [`number`]    | The maximum number of results to return.
```
Comment (exopsRamba) [inline content/millennium/r4/financial/charge-item.md:90]: nit: Let's put the paging parameter at the bottom of the list.

File: content/millennium/r4/financial/charge-item.md [hunk 5]
```diff
@@ -83,13 +83,18 @@ Search for ChargeItems that meet supplied query parameters:
...
+  `-status`            | No              | [`token`]     | The status of the encounter. Example: `billable`
 
 Notes:
 
 - When searching by `_id`, only a single ChargeItem id can be provided.
+- `_count` and `-status` must be provided with context.
```
Comment (exopsRamba) [inline content/millennium/r4/financial/charge-item.md:96]: This seems to imply that _count is a required parameter when context is passed. We can probably just remove _count from this line.

If we feel it's necessary to specify where _count can be used we could separate it into a line specifying `'_count' cannot be provided with '_id'`

File: content/millennium/r4/financial/charge-item.md (hunk 3 above)
Comment (alexbeckner1) [reply content/millennium/r4/financial/charge-item.md:91]: +1 to See notes

File: content/millennium/r4/financial/charge-item.md (hunk 4 above)
Comment (alexbeckner1) [reply content/millennium/r4/financial/charge-item.md:90]: +1

File: content/millennium/r4/financial/charge-item.md (hunk 5 above)
Comment (alexbeckner1) [reply content/millennium/r4/financial/charge-item.md:96]: +1 to removing _count from this. I'd say we only add the other note for _count if we are doing it in other resources.

File: content/millennium/r4/financial/charge-item.md (hunk 3 above)
Comment (Anurag-Balannagari) [reply content/millennium/r4/financial/charge-item.md]: updated here https://github.com/cerner/fhir.cerner.com/pull/665/commits/66cb49e49a70cc2e124fb6348991af2240402aa6#diff-16d4f8f74626c62d57a8b71d6be699f60e72fffed8636bc32e6654d4252b3ebbR90

File: content/millennium/r4/financial/charge-item.md (hunk 4 above)
Comment (Anurag-Balannagari) [reply content/millennium/r4/financial/charge-item.md]: updated here https://github.com/cerner/fhir.cerner.com/pull/665/commits/66cb49e49a70cc2e124fb6348991af2240402aa6#diff-16d4f8f74626c62d57a8b71d6be699f60e72fffed8636bc32e6654d4252b3ebbR91

File: content/millennium/r4/financial/charge-item.md (hunk 5 above)
Comment (Anurag-Balannagari) [reply content/millennium/r4/financial/charge-item.md]: removed _count here https://github.com/cerner/fhir.cerner.com/pull/665/commits/66cb49e49a70cc2e124fb6348991af2240402aa6#diff-16d4f8f74626c62d57a8b71d6be699f60e72fffed8636bc32e6654d4252b3ebbR96

**Comment by Anurag-Balannagari** [comment]
> This should say "The status of the charge item. Example: `billable`"

updated here https://github.com/cerner/fhir.cerner.com/pull/665/commits/d908e31bea82197ef513ba04c1531dbfe7ceaa58#diff-16d4f8f74626c62d57a8b71d6be699f60e72fffed8636bc32e6654d4252b3ebbR90

**Comment by alexbeckner1** [comment]
> > This should say "The status of the charge item. Example: `billable`"
> 
> updated here [d908e31#diff-16d4f8f74626c62d57a8b71d6be699f60e72fffed8636bc32e6654d4252b3ebbR90](https://github.com/cerner/fhir.cerner.com/commit/d908e31bea82197ef513ba04c1531dbfe7ceaa58#diff-16d4f8f74626c62d57a8b71d6be699f60e72fffed8636bc32e6654d4252b3ebbR90)

**Comment by Anurag-Balannagari** [comment]
> > This should say "The status of the charge item. Example: `billable`"
> 
> updated here [d908e31#diff-16d4f8f74626c62d57a8b71d6be699f60e72fffed8636bc32e6654d4252b3ebbR90](https://github.com/cerner/fhir.cerner.com/commit/d908e31bea82197ef513ba04c1531dbfe7ceaa58#diff-16d4f8f74626c62d57a8b71d6be699f60e72fffed8636bc32e6654d4252b3ebbR90)

updated here https://github.com/cerner/fhir.cerner.com/pull/665/commits/488fe50448eb0b4debde825674864a2c5cca3eb7#diff-16d4f8f74626c62d57a8b71d6be699f60e72fffed8636bc32e6654d4252b3ebbR90

File: content/millennium/r4/financial/charge-item.md [hunk 6]
```diff
@@ -106,6 +111,17 @@ Notes:
 <%= headers status: 200 %>
 <%= json(:r4_chargeitem_bundle) %>
 
+### Example - Search by Context
```
Comment (alexbeckner1) [inline content/millennium/r4/financial/charge-item.md:114]: Should we get rid of the Search by Id example all together since it is pretty much the exact same as a Retreive by Id?

Comment (exopsRamba) [reply content/millennium/r4/financial/charge-item.md:114]: +1

File: content/millennium/r4/financial/charge-item.md [hunk 7]
```diff
@@ -83,13 +83,18 @@ Search for ChargeItems that meet supplied query parameters:
...
 ### Parameters
 
- Name                  | Required?  | Type          | Description
------------------------|------------|---------------|-------------------------------------------------------
- `_id`                 | Yes        | [`token`]     | The logical resource id associated with the ChargeItem.
+ Name                  | Required?       | Type          | Description
+-----------------------|-----------------|---------------|-------------------------------------------------------
+ `_id`                 | This or context | [`token`]     | The logical resource id associated with the ChargeItem.
+ `context`             | This or _id     | [`token`]     | Encounter associated with event.
+ `-status`             | see notes       | [`token`]     | The status of the ChargeItem. Example: `billable`
```
Comment (exopsRamba) [inline content/millennium/r4/financial/charge-item.md:90]: nit: `see notes` should be capitalized. i.e. `See notes`

File: content/millennium/r4/financial/charge-item.md (hunk 6 above)
Comment (Anurag-Balannagari) [reply content/millennium/r4/financial/charge-item.md]: removed it here https://github.com/cerner/fhir.cerner.com/pull/665/commits/52ec1eb5766e51bfbe059e26fbe147fab674940c#diff-16d4f8f74626c62d57a8b71d6be699f60e72fffed8636bc32e6654d4252b3ebbL103-L114

File: content/millennium/r4/financial/charge-item.md (hunk 7 above)
Comment (Anurag-Balannagari) [reply content/millennium/r4/financial/charge-item.md]: updated here https://github.com/cerner/fhir.cerner.com/pull/665/commits/52ec1eb5766e51bfbe059e26fbe147fab674940c#diff-16d4f8f74626c62d57a8b71d6be699f60e72fffed8636bc32e6654d4252b3ebbR90

**Review by exopsRamba — APPROVED** [review]

**Review by ahamdany — APPROVED** [review]

**Review by alexbeckner1 — APPROVED** [review]

File: content/millennium/r4/financial/charge-item.md (hunk 6 above)
Comment (alexbeckner1) [reply content/millennium/r4/financial/charge-item.md]: Thanks for making that change.

File: content/millennium/r4/financial/charge-item.md [hunk 8]
```diff
@@ -83,23 +83,28 @@ Search for ChargeItems that meet supplied query parameters:
...
+-----------------------|-----------------|---------------|-------------------------------------------------------
+ `_id`                 | This or context | [`token`]     | The logical resource id associated with the ChargeItem.
+ `context`             | This or _id     | [`token`]     | Encounter associated with event.
+ `-status`             | See notes       | [`token`]     | The status of the ChargeItem. Example: `billable`
+ [`_count`]            | No              | [`number`]    | The maximum number of results to return.
 
 Notes:
 
 - When searching by `_id`, only a single ChargeItem id can be provided.
+- `-status` must be provided with context.
```
Comment (MadhurRaj) [inline content/millennium/r4/financial/charge-item.md:96]: put context in the same format as status

`context`

File: content/millennium/r4/financial/charge-item.md [hunk 9]
```diff
@@ -83,23 +83,28 @@ Search for ChargeItems that meet supplied query parameters:
...
+- `-status` must be set to `billable`.
```
Comment (MadhurRaj) [inline content/millennium/r4/financial/charge-item.md:97]: we'll need to document that `context` has to reference an Encounter.

File: content/millennium/r4/financial/charge-item.md (hunk 8 above)
Comment (Anurag-Balannagari) [reply content/millennium/r4/financial/charge-item.md]: updated here https://github.com/cerner/fhir.cerner.com/pull/665/commits/7dcf1473cb198a27c98dcb618252085c6dcfb703#diff-16d4f8f74626c62d57a8b71d6be699f60e72fffed8636bc32e6654d4252b3ebbR96

File: content/millennium/r4/financial/charge-item.md (hunk 9 above)
Comment (Anurag-Balannagari) [reply content/millennium/r4/financial/charge-item.md:97]: added here https://github.com/cerner/fhir.cerner.com/pull/665/commits/7dcf1473cb198a27c98dcb618252085c6dcfb703#diff-16d4f8f74626c62d57a8b71d6be699f60e72fffed8636bc32e6654d4252b3ebbR98

Comment (Anurag-Balannagari) [reply content/millennium/r4/financial/charge-item.md:97]: updated here to "context must reference an Encounter."
https://github.com/cerner/fhir.cerner.com/pull/665/commits/bab355ecb70a4975136c483510c69526a047c165#diff-16d4f8f74626c62d57a8b71d6be699f60e72fffed8636bc32e6654d4252b3ebbR98

**Review by MadhurRaj — APPROVED** [review]

**Comment by Zonglin123** [comment]
Screen Shot 2021-06-18 at 12 51 20 PM
Screen Shot 2021-06-18 at 12 51 31 PM
Screen Shot 2021-06-18 at 12 51 52 PM
